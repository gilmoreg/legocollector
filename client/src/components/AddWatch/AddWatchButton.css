.AddWatchButton {
  border: 10px solid red;
  border-radius: 50%;
  background-color: red;
  width: 50px;
  height: 50px;
  position: fixed;
  bottom: 50px;
  right: 50px;
}

.AddWatchButton span {
  color: white;
  font-size: 40px;
  position: relative;
  /* TODO center more programmatically */
  top: -12px;
  left: -3px;
}