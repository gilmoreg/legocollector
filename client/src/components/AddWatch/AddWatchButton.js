import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const StyledButton = styled.button`
  border: 10px solid #FF7F66;
  border-radius: 50%;
  background-color: #FF7F66;
  width: 50px;
  height: 50px;
  position: fixed;
  bottom: 50px;
  right: 25px;
  z-index: 0;
`;

const StyledSpan = styled.span`
  color: white;
  font-size: 40px;
  position: relative;
  top: -12px;
  left: -3px;
  content: "+";
`;

const AddWatchButton = props => (
  <StyledButton onClick={props.openModal}>
    <StyledSpan>+</StyledSpan>
  </StyledButton>
);

AddWatchButton.propTypes = {
  openModal: PropTypes.func.isRequired,
};

export default AddWatchButton;
