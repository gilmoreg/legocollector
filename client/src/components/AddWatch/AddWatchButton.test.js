import React from 'react';
import { shallow } from 'enzyme';
import AddWatchButton from './AddWatchButton';

it('renders without crashing', () => {
  const wrapper = shallow(
    <AddWatchButton openModal={() => {}} />);
  expect(wrapper).toMatchSnapshot();
});
