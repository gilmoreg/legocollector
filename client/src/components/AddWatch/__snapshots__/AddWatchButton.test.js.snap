// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddWatchButton
    openModal={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": <styled.span>
        +
      </styled.span>,
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": "+",
      },
      "ref": null,
      "rendered": "+",
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": <styled.span>
          +
        </styled.span>,
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": "+",
        },
        "ref": null,
        "rendered": "+",
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
